src/Messenger/LegacyMessenger.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\service_container\LegacyMessenger.
+ */
+
+namespace Drupal\service_container\Messenger;
+
+/**
+ * Class that manage the messages in Drupal.
+ *
+ * @codeCoverageIgnore
+ */
+class LegacyMessenger implements MessengerInterface {
```

Comment:
Add a __construct taking the Drupal 7 service as first argument, also add a protected value for the drupal 7 service - similar to the integration test for the legacy service. ($this->drupal7 = $drupal7_service)